src/app/recipes/recipes-cards/recipes-cards.component.ts

```diff
@@ -23,4 +28,20 @@ export class RecipesCardsComponent implements OnInit {
 			RecipesSelectors.selectDataState
 		);
 	}
+
+	public handleTextSearchInput(value: Event): void {
+		this.searchText = (<HTMLInputElement>value.target).value;
+	}
+
+	public selectFileToSort(event: Event): void {
+		this.sortCategory = (<HTMLInputElement>event.target).value as keyof Recipe;
+	}
+
+	public changeSortDirection(): void {
+		if (this.sortDirection === 'asc') {
+			this.sortDirection = 'desc';
+			return;
+		}
+		if (this.sortDirection === 'desc') this.sortDirection = 'asc';
```

Comment:
`this.sortDirection = this.sortDirection === YourEnum.Desc ? YourEnum.Asc : YourEnum.Desc`